src/parcsr_ls/par_ilu.c

```diff
@@ -12,73 +12,131 @@
  *****************************************************************************/
 
 #include "_hypre_parcsr_ls.h"
+#include "_hypre_utilities.hpp"
 #include "par_ilu.h"
 #include <assert.h>
 
 /* Create */
 void *
 hypre_ILUCreate()
 {
-   hypre_ParILUData                    *ilu_data;
-
-   ilu_data                            = hypre_CTAlloc(hypre_ParILUData,  1, HYPRE_MEMORY_HOST);
+   hypre_ParILUData                       *ilu_data;
+
+   ilu_data                               = hypre_CTAlloc(hypre_ParILUData,  1, HYPRE_MEMORY_HOST);
+
+#ifdef HYPRE_USING_CUDA
+   ((ilu_data) -> matL_des)               = NULL;
```

Comment:
We should use `hypre_ParILUDataMatLMatrixDescription(ilu_data)` macro, and in many other places. I know there are a lot ...